pex/pex.py

```diff
@@ -94,6 +94,18 @@ def _extras_paths(cls):
     extras_paths = filter(None, makefile.get('EXTRASPATH', '').split(':'))
     for path in extras_paths:
       yield os.path.join(standard_lib, path)
+    # macOS Sierra support
+    framework = sysconfig.get_config_var("PYTHONFRAMEWORK")
+    if not framework:
+      raise StopIteration()
+    third_party = os.path.join("/Library", framework, sys.version[:3], "site-packages")
+    yield third_party
+    extras_pth = os.path.join(third_party, 'Extras.pth')
+    if not os.path.exists(extras_pth):
+      raise StopIteration()
+    with open(extras_pth, 'r') as fd:
+      for line in fd.readlines():
+        yield line.strip()
```

Comment:
I see a few problems with this approach for backward and forward compatibility:

1) .pth files can contain comments, which aren't handled here
2) .pth files can contain relative paths, which wouldn't make sense to directly yield
3) .pth files can contain paths that don't exist, which isn't checked here
4) technically, .pth files that begin with `import` can contain executable code, which isn't handled here (and probably should not be - but at least should detect and avoid emitting raw code)

so, if we must rely on our own .pth file parsing it'd probably be good to extract a separate, general purpose .pth file parser function into `pex.util` and test that separately.

you can probably lift much of the required logic directly from the `site` module's .pth handling implementation here: https://github.com/kwlzn/python-sources/blob/master/Python-2.7.12/Lib/site.py#L131